custom_components/bhyve/switch.py

```diff
@@ -515,6 +525,27 @@ def is_on(self):
         """Return the status of the sensor."""
         return self._is_on
 
+    async def set_smart_watering_soil_moisture(self, percentage):
+        """Set the soil moisture percentage for the zone."""
+        if self._attrs[ATTR_SMART_WATERING_ENABLED]:
+            landscape = await self._bhyve.get_landscape(self._device_id, self._zone_id)
+            if landscape is not None:
+                _LOGGER.debug("Landscape data %s", landscape)
+                update_json = {}
+                to_remove = ['available_water', 'created_at', 'distribution_uniformity', 'landscape_coefficient', 'plant_available_water', 'updated_at']
+
+                # Remove properties from update json
+                for item in landscape:
+                    if item not in to_remove:
+                        update_json[item] = landscape[item]
+                # Add moisture level to update json
+                update_json['current_water_level'] = landscape['replenishment_point'] + ((percentage * (landscape['field_capacity_depth'] - landscape['replenishment_point'])) / 100.0)
```

Comment:
Can you please explain how this works? What are the `replenishment_point` and `field_capacity_depth` attributes?

Reply:
When you define smart watering on my controller it asks for a bunch of information including soil type (loam, clay, sand), plant types (shrubs, grass) etc which it calculates data from.

`replenishment_point` is the level it considers to be no/0% moisture, and
`field_capacity_depth` is when it would be full/100% moisture.

Could write it something like this?
```
landscape_moisture_level_0 = landscape['replenishment_point']
landscape_moisture_level_100 = landscape['field_capacity_depth'] 
update_json['current_water_level'] = landscape_moisture_level_0 + ((percentage * (landscape_moisture_level_100 - landscape_moisture_level_0)) / 100.0)
```

Here is the json for a landscape/zone returned by `get_landscape()`

```
    {
        "allowable_soil_acc": 0.26,
        "application_rate": 0.8665633916854858,
        "available_water": 0.14,
        "created_at": "2021-10-26T09:25:06.371Z",
        "device_id": "6177c75f4f0c6f063b848d36",
        "distribution_uniformity": null,
        "drought_factor": 1,
        "efficiency": 0.75,
        "field_capacity": 0.32,
        "field_capacity_depth": 1.92,
        "id": "6177c9724f0cf51668ed80d4",
        "infiltration_rate": 0.2,
        "landscape_coefficient": 0.54,
        "max_allowable_depletion": 0.5,
        "max_runtime": 23,
        "micro_climate": 0.9,
        "monthly_eto": [
            3.65868997574,
            3.12609004974,
            2.42949008942,
            1.59974002838,
            1.02868998051,
            0.733566999435,
            0.804527997971,
            1.10535001755,
            1.80235004425,
            2.50878000259,
            3.03544998169,
            3.43576002121
        ],
        "permanent_wilting_point": 0.18,
        "plant_available_water": 0.8399999999999999,
        "plant_factor": 0.6,
        "pwp_depth": 1.08,
        "rainfall_efficiency": 0.8,
        "readily_available_water": 0.41999999999999993,
        "replenishment_point": 1.5,
        "root_depth": 6,
        "scheduling_multiplier": 1.1764705882352942,
        "station": 1,
        "updated_at": "2022-01-25T20:40:40.070Z"
    },
```

Reply:
Thats super helpful. The changes you suggested (with some inline comments) would be perfect.